src/getDefaultHostingSite.ts

```diff
@@ -0,0 +1,19 @@
+import * as logger from "./logger";
+import { getFirebaseProject } from "./management/projects";
+import * as _ from "lodash";
+
+/**
+ * Tries to determine the default hosting site for a project, else falls back to projectId.
+ * @param {Object} options The command-line options object
+ * @returns {Promise<String>} The hosting site ID
+ */
+export async function getDefaultHostingSite(options: any): Promise<string> {
+  const project = await getFirebaseProject(options.project);
+  if (!_.has(project, "resources.hostingSite")) {
```

Comment:
I would like to discourage the use of lodash here. The returned value from `getFirebaseProject` is well defined (`FirebaseProjectMetadata`). Could use more typescripty ways of checking for these that allow the types to be relied upon?

For `has` (untested):

```suggestion
  if (project.resources?.hostingSite) {
```